Search CRUD topics

jamesm-macrometa opened the pull request:
This PR adds API, Python, JS, and CLI instructions for creating a Search View.

Reviewers:  Please take a look at the JavaScript example.  It seems to be in the right direction, and it runs without errors, but for some reason it does not actually create a Search View like the Python example does.  ChatGPT has run out of ideas troubleshooting and suggested I contact Macrometa support.

jamesm-macrometa committed:
Update create-search-views.md

jamesm-macrometa committed:
Update create-search-views.md

jamesm-macrometa committed:
Update create-search-views.md

jamesm-macrometa committed:
Update create-search-views.md

jamesm-macrometa committed:
Update create-search-views.md

jamesm-macrometa committed:
Update create-search-views.md

koshy-thomas commented on docs/search/views/create-search-views.md:
```diff
@@ -24,3 +30,145 @@ Follow these instructions to create a new search view using the GDN console web
+
+  # Create test view on demo collection with include-all-fields
+  gdnsl view create test --collection demo --include-all-fields
+
```
we should also add gdnsl example command with primary sort field..

gdnsl view create test --collection demo --field "name:text_en,text_fr" --field "age:text_en" --primary-sort-field "name:asc"

diana-macrometa replied:
I generally just link to the CLI topic or heading with the info needed, cuts down on the maintenance overhead. If this is wrong, though, we should also review the source CLI topic, @jamesm-macrometa .

koshy-thomas commented on docs/search/views/create-search-views.md:
````diff
@@ -24,3 +30,145 @@ Follow these instructions to create a new search view using the GDN console web
+      --track-list-positions boolean  The flag determines whether or not values in a lists should be treated separate (default: false).
+      --include-all-fields boolean    The flag determines whether or not to index all fields on a particular level of depth (default: false).
+      --fabric                 Name of the fabric to use.
+```
````
--primary-sort-field  string  The flag accepts the fields and the sort order by which the collection will be sorted. Format "fieldName:sortOrder (either "asc" or "desc")".
Example: --primary-sort-field "name:asc".

koshy-thomas commented on docs/search/views/create-search-views.md:
```diff
@@ -24,3 +30,145 @@ Follow these instructions to create a new search view using the GDN console web
+search_view_name = "example_search_view"
+collection_name = "your_collection_name"
+properties = {
+    "<COLLECTION NAME>": { # Change this to the name of the collection to use with the search view.
```
collection_name is already wrapped we need to remove this extra wrapping.. let's have a call @jamesm-macrometa when you are online we will also add primary sort example here too and the same for Javascript too and will give you the working pyC8, jsc8 codes

jamesm-macrometa committed:
Update create-search-views.md

jamesm-macrometa committed:
fix link

jamesm-macrometa committed:
Merge pull request #840 from Macrometacorp/update-branch

Update branch

jamesm-macrometa committed:
Update create-search-views.md

jamesm-macrometa committed:
Merge branch 'search-CRUD-topics' of https://github.com/Macrometacorp/docs into search-CRUD-topics

jamesm-macrometa commented:
@brandonLyonMM @diana-macrometa Any ideas why this branch is failing to build?  When I run the test on my own machine, it builds without any issues.

diana-macrometa committed:
Merge branch 'main' into search-CRUD-topics

diana-macrometa committed:
Merge branch 'search-CRUD-topics' of https://github.com/Macrometacorp/docs into search-CRUD-topics

diana-macrometa commented:
@jamesm-macrometa, did you do `yarn run build` to check for broken links?

brandonLyonMM commented:
> @brandonLyonMM @diana-macrometa Any ideas why this branch is failing to build? When I run the test on my own machine, it builds without any issues.

The error message from our CI-CD builder is:

> Error: Docs markdown link couldn't be resolved: (../../CLI/search-views-cli.md) in "/vercel/path0/docs/search/views/create-search-views.md" for version current

diana-macrometa commented:
@brandonLyonMM, I actually cannot see the Vercel error messages.

jamesm-macrometa commented:
@brandonLyonMM @diana-macrometa This is very strange, the link works perfectly fine for me in VSCode and does not return any error when I run `yarn run build` on my system.  Any ideas how to fix this?

diana-macrometa commented:
Are you sure you are on the correct, updated branch? Because I got an error on that branch as well.

jamesm-macrometa committed:
Update create-search-views.md

jamesm-macrometa committed:
Merge branch 'search-CRUD-topics' of https://github.com/Macrometacorp/docs into search-CRUD-topics

jamesm-macrometa commented:
> Are you sure you are on the correct, updated branch? Because I got an error on that branch as well.

I think my issues were caused by a broken link in Main, opened https://github.com/Macrometacorp/docs/pull/842 to fix it.

jamesm-macrometa committed:
Merge pull request #843 from Macrometacorp/main

Fix links in PR

jamesm-macrometa committed:
Update create-search-views.md

jamesm-macrometa commented:
@brandonLyonMM @diana-macrometa For some reason the link I added didn't return an error for my local build, but when I removed `.md` from the link it works now.  Thanks for the help!

jamesm-macrometa commented:
Created https://github.com/Macrometacorp/docs/pull/844/files to address Koshy's comments re: the CLI commands.
